tests/integration/support/steps.go

```diff
@@ -0,0 +1,274 @@
+package support
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/optimizely/go-sdk/optimizely/event"
+
+	"github.com/DATA-DOG/godog/gherkin"
+	"github.com/optimizely/go-sdk/optimizely/entities"
+	"github.com/optimizely/go-sdk/tests/integration/models"
+	"github.com/optimizely/go-sdk/tests/integration/optlyplugins"
+	"gopkg.in/yaml.v3"
+)
+
+// ScenarioCtx holds both apiOptions and apiResponse for a scenario
+type ScenarioCtx struct {
+	apiOptions  models.RequestParams
+	apiResponse models.ResponseParams
+	wrapper     ClientWrapper
+}
+
+// TheDatafileIs represents a step in the feature file
+func (c *ScenarioCtx) TheDatafileIs(datafileName string) error {
+	c.apiOptions.DatafileName = datafileName
+	return nil
+}
+
+// ListenerIsAdded represents a step in the feature file
+func (c *ScenarioCtx) ListenerIsAdded(numberOfListeners int, ListenerName string) error {
+	if c.apiOptions.Listeners == nil {
+		c.apiOptions.Listeners = make(map[string]int)
+	}
+	c.apiOptions.Listeners[ListenerName] = numberOfListeners
+	return nil
+}
+
+// IsCalledWithArguments represents a step in the feature file
+func (c *ScenarioCtx) IsCalledWithArguments(apiName string, arguments *gherkin.DocString) error {
+	// Check if its a new scenario
```

Comment:
Is this how we maintain "sessions"? You should have the wrapper keep track of what the scenario is as a property and compare that instead